go.mod

```diff
@@ -21,6 +21,7 @@ require (
 	github.com/emicklei/go-restful v2.15.0+incompatible // indirect
 	github.com/evanphx/json-patch v4.9.0+incompatible
 	github.com/fatih/structs v1.1.0 // indirect
+	github.com/felixge/httpsnoop v1.0.1
```

Comment:
can we avoid this new library? if not, what licenses is in under?

Reply:
httpsnoop is (at the time of that version) licensed under [the MIT license](https://github.com/felixge/httpsnoop/blob/master/LICENSE.txt)

For the rationale of why I'm using it, check out https://github.com/felixge/httpsnoop#why-this-package-exists.  Originally, I did not use an extra lib and attempted to simply wrap the ResponseWriter, but I ended up causing a slew of errors due to not implementing additional interfaces and not implementing some of the other edge cases mentioned in that link.